src/chainparamsbase.h

```diff
@@ -31,6 +32,13 @@ class CBaseChainParams
 };
 
 /**
+ * Creates and returns a std::unique_ptr<CBaseChainParams> of the chosen chain.
+ * @returns a CBaseChainParams* of the chosen chain.
```

Comment:
In commit "Chainparams: Use a regular factory for creating chainparams"

Technically returns a unique_ptr